Better pager

zeronetscript opened the pull request:
these two patch based on my previously coffeelint work, it supports better pager. 

page links may be less than it should be , when applyPagerdata called, whole page didn't layout completely so it needs further work. I'm not good at html so I just create this pull, hopes others can improve it

Commit:
replace all tab to 2 spaces

Commit:
quiet coffeelint

HelloZeroNet commented:
hi, thanks for the contribution, but i don't really have time to review/test the changes, so please create a new fork/demo site and I will link to this as "improved blog" on zeroblog

Commit:
get total post when query posts

when selecting posts, select count(*) from post as first row
so we can do better page display

Commit:
better add blog page support

previously only show prev/next, hard to jump. this improve it by:
support jump to first/last page
show current page and neighbour page links (instead of prev/next)
if not all next pages can showup, ... added to last link to notice user

zeronetscript commented:
I've create a demo blog for review here:
http://127.0.0.1:43110/19k2nhubaYsDJNPHL6iNs3D2K9Mgj66P4R
